internal/parse-options.js

```diff
@@ -1,11 +1,41 @@
-// parse out just the options we care about so we always get a consistent
-// obj with keys in a consistent order.
-const opts = ['includePrerelease', 'loose', 'rtl']
-const parseOptions = options =>
-  !options ? {}
-  : typeof options !== 'object' ? { loose: true }
-  : opts.filter(k => options[k]).reduce((o, k) => {
-    o[k] = true
-    return o
-  }, {})
-module.exports = parseOptions
+const var1 = Object.freeze({ includePrerelease: true, loose: true, rtl: true });
+const var2 = Object.freeze({ includePrerelease: true, loose: true });
+const var3 = Object.freeze({ includePrerelease: true, rtl: true });
+const var4 = Object.freeze({ includePrerelease: true });
+const var5 = Object.freeze({ loose: true, rtl: true });
+const var6 = Object.freeze({ loose: true });
+const var7 = Object.freeze({ rtl: true });
+const emptyOpts = Object.freeze({});
```

Comment:
typically freezing objects makes them much slower, so it's worth benchmarking this.

Also, all of these should be `__proto__: null` so lookups can be faster.

Reply:
+1 on trying out `Object.create(null)`, but I wonder if trying out making this a `Map`, or maybe even using [Bitmasks](https://levelup.gitconnected.com/bit-masking-in-javascript-831eb26f04a3) could speed it up further?

Reply:
Yeah, freezing has never changed performance in anything I've ever tried, unfortuantely.

Honeslty, I think that the way forward in optimizing things is not going to be down to tricks like freezing or looking for the fastest LRU cache; it's going to be overall algorithmic changes to the code.

Reply:
I think @ljharb 's point was more about using `Object.freeze` (the fn itself) makes the code slower, not faster (because the freezing itself is an operation).

Reply:
@ljharb I run `parse-options` benchmark and `satisfies` benchmark, I didn't notice any perf drawback.

About `__proto__`, do you mean `Object.freeze({ __proto__: null, rtl: true })`?

@kurtextrem Creating the object with `Object.create(null)` is actually fast ([ref](https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v19.md#object- creation) ), but since I stop to create new objects, I don't need it.

About the bitmask, sure I could be a lot faster, but I didn't want to introduce any significant changes, but it might be worth exploring in another PR with significant changes in mind.

Reply:
Yes, that's what I mean, so it doesn't look up to Object.prototype.

Reply:
If I put `__proto__: null`, it breaks the tests of strictSame:

![image](https://user-images.githubusercontent.com/12551007/229262350-2159c02a-7431-4753-b1df-56a182cb8b26.png)

Change the tests or don't add `__proto__`?

Reply:
Try benchmarking before changing the tests, but in a naive test I did creating Objects with `__proto__:null` was actually slower, but I don't really know why (which is why I did not send the benchmark)

Reply:
@kurtextrem like https://github.com/npm/node-semver/pull/528#discussion_r1155113948, if we do it inline it can be slower, so as not to break the tests, I think it's better not to include `__proto__: null`.

Reply:
It would be fine to change the tests here, to be clear. I'm very surprised doing it inline is slower - in which engines is that true?

Note that we're not just concerned with node here, this package is used in browsers as well.

Reply:
@ljharb The benchmark site I've used (from the other thread) doesn't work for me in Firefox, so I can only say it's definitely much slower in Chromium, Safari and Node (that caught my be surprise too, I assume '__proto__' has a special treatment during creation if not done with `Object.create(null)`, thus making it slower than the standard object). It's even slower than `Object.freeze/seal`.

Reply:
screenshot, because maybe not everyone has access to a Mac following this discussion: 
<img width="1318" alt="image" src="https://user-images.githubusercontent.com/502146/229438225-2d29870b-04cb-4327-add4-5e5de3d225f1.png">

Reply:
@kurtextrem The main problem is if we were returning the object since we run it once, the op/s is not a problem.

It will be a problem if it slows down the property access but I don' think that is the case.